EventGenerator/src/RPCGun_module.cc

```diff
@@ -152,14 +154,12 @@ namespace mu2e {
       const PhysicsParams& gc = *GlobalConstantsHandle<PhysicsParams>();
       double weight = 0.;
       double tau = part->endProperTime() / gc.getParticleLifetime(part->pdgId());
-
       while(part->parent().isNonnull()) { //while particle has a parent which is not null
         part = part->parent();
-        if ( abs(part->pdgId()) == PDGCode::pi_plus ) { //if pion
+        if ( abs(part->pdgId()) == 211 ) { //if pion
```

Comment:
Two things:

1. Always use std::abs, never bare abs.  It leaves you vulnerable to changes in behaviour if transitively included headers change.
2.  Always use symbolic names.  I excpect you can use std::abs(PDGCode::pi_plus).

Reply:
yes, was part of a test due to wierd result, wanted to make sure it was parsing.....fixed now